src/sentry/api/base.py

```diff
@@ -450,6 +454,18 @@ def dispatch(self, request: Request, *args, **kwargs) -> Response:
                     span.set_data("SENTRY_API_RESPONSE_DELAY", settings.SENTRY_API_RESPONSE_DELAY)
                     time.sleep(settings.SENTRY_API_RESPONSE_DELAY / 1000.0 - duration)
 
+        # Only enforced in dev environment
+        if settings.ENFORCE_PAGINATION:
+            if request.method.lower() == "get":
+                # Response can either be Response or HttpResponse, check if it's value an array
+                if hasattr(self.response, "data") and isinstance(self.response.data, list):
+                    # if not paginated and not in  settings.SENTRY_API_PAGINATION_ALLOWLIST, raise error
+                    if (
+                        not self.has_pagination(self.response)
+                        and handler.__self__.__class__.__name__
+                        not in settings.SENTRY_API_PAGINATION_ALLOWLIST
+                    ):
```

Comment:
nit: I think we want to change the order of operations here, and check that it's not in the allowlist first, then do the actual checking. This way in the future if the logic is more complex and expensive, we don't need to worry about it being a performance problem.